data/rulesets.js

```diff
@@ -737,7 +737,7 @@ let BattleFormats = {
 			for (const pokemon of this.getAllPokemon()) {
 				const item = pokemon.getItem();
 				// this.canMegaEvo check is for Rayquaza.
-				if (item.megaStone || this.canMegaEvo(pokemon) || item.onPrimal || item.zMove) {
+				if (item.megaStone || this.canMegaEvo(pokemon) || item.onPrimal || item.zMove || (pokemon.transformed && (pokemon.template.isMega || pokemon.template.isPrimal || pokemon.template.forme === "Ultra"))) {
```

Comment:
This only runs at the start of the battle, so it has no effect on Ditto.

Reply:
thanks for the feedback @urkerab, much appreciated